pip/pip-294.md

```diff
@@ -0,0 +1,56 @@
+# Background knowledge
+
+Load balance module in Pulsar broker rely on zk to store and synchronize metadata about load. Every broker will upload its `LocalBrokerData` to zk, and leader broker will retrieve all `LocalBrokerData` from zk ,generate all `BundleData` from each `LocalBrokerData`, and update all `BundleData` to zk. 
+
+
+# Motivation
+
+As every bundle in the cluster corresponds to a zk node, it is common that there are thousands of zk nodes in a cluster, which results into thousands of read/update operations to zk. This will cause a lot of pressure on zk.
+
+**As All Load Shedding Algorithm pick bundles from top to bottom based on throughput/msgRate, bundles with low throughput/msgRate are rarely be selected for shedding. So there is no need to update these bundleData to zk frequently.**
+
+
+# Goals
+
+Filter out bundles with low throughput/msgRate, and do not update these bundles to zk frequently to reduce the pressure on zk.
+
+
+# High Level Design
+
+Filter out bundles with low throughput/msgRate when leader update bundleData to zk.
+
+
+# Detailed Design
+
+## Design & Implementation Details
+Add throughput/msg-based bundle report control in the `ExtensibleLoadManager` and `ModularLoadManagerImpl`.
```

Comment:
No, there is no such expiration mechanism. We may need to add such expiration mechanism for both `ExtensibleLoadManager` and `ModularLoadManagerImpl`.
But should we implement it in this PR?

Reply:
Ok, got it. Instead of introducing an expiration mechanism. Can we just contain such bundles by comparing the current load and the last uploaded load?

Reply:
How about: If the current load is less than 1/10 of the previous load, we will trigger its update.
- Comparing mechanism  we need to determine the threshold when should we trigger the updates.
- Expiration mechanism  we need to determine the frequency of update small bundles. This solutions may still cause zk latency rise sharply periodically.
We also need to add such expiration or comparing mechanism to topK mechanism, which is why i wonder we should implement it in this PR.

Reply:
The extensible load manager has the `hit counter` logic under `loadBalancerSheddingConditionHitCountThreshold` for the bursty traffic.